### PR TITLE
Support partial parsing when cache is disabled

### PR DESCRIPTION
**Pull request by tatiana**
Partial parsing support was introduced in #800 and improved in #904 (caching). However, as the caching layer was introduced, we removed support to use partial parsing if the cache was disabled.

This PR solves the issue.

Fix: #1041

### CODE REVIEW

File: cosmos/dbt/graph.py [hunk 1]
```diff
@@ -424,18 +428,21 @@ def load_via_dbt_ls_without_cache(self) -> None:
...
+            logger.debug(f"Content of the dbt project dir {project_path}: `{os.listdir(project_path)}`")
             tmpdir_path = Path(tmpdir)
 
             create_symlinks(project_path, tmpdir_path, self.render_config.dbt_deps)
 
-            if self.project.partial_parse and self.cache_dir:
-                latest_partial_parse = cache._get_latest_partial_parse(project_path, self.cache_dir)
+            latest_partial_parse = None
+            if self.project.partial_parse:
+                if self.should_use_partial_parse_cache() and self.cache_dir:
```
Comment (pankajastro) [inline cosmos/dbt/graph.py:438]: Do we still need check `self.cache_dir` here, considering that `should_use_partial_parse_cache` already performs check `bool(self.cache_dir)`?

**Review by pankajastro — APPROVED** [review]
lgtm

File: cosmos/dbt/graph.py [hunk 2]
```diff
@@ -424,18 +428,21 @@ def load_via_dbt_ls_without_cache(self) -> None:
...
+                    latest_partial_parse = cache._get_latest_partial_parse(project_path, self.cache_dir)
+                else:
+                    latest_partial_parse = get_partial_parse_path(project_path)
+
+            if latest_partial_parse is not None and latest_partial_parse.exists():
                 logger.info("Partial parse is enabled and the latest partial parse file is %s", latest_partial_parse)
-                if latest_partial_parse is not None:
-                    cache._copy_partial_parse_to_project(latest_partial_parse, tmpdir_path)
+                cache._copy_partial_parse_to_project(latest_partial_parse, tmpdir_path)
```
Comment (pankajkoti) [inline cosmos/dbt/graph.py:445]: I am wondering if we need another check here to see if cache is enabled and only then copy it?

File: cosmos/dbt/graph.py [hunk 3]
```diff
@@ -470,9 +477,9 @@ def load_via_dbt_ls_without_cache(self) -> None:
                 self.nodes = nodes
                 self.filtered_nodes = nodes
 
-            if self.project.partial_parse and self.cache_dir:
+            if self.should_use_partial_parse_cache():
                 partial_parse_file = get_partial_parse_path(tmpdir_path)
-                if partial_parse_file.exists():
+                if partial_parse_file.exists() and self.cache_dir:
```
Comment (pankajkoti) [inline cosmos/dbt/graph.py:482]:
```suggestion
                if partial_parse_file.exists():
```
based on Pankaj's comment above that we already check this prior to getting here.

File: cosmos/dbt/graph.py (hunk 1 above)
Comment (pankajkoti) [reply cosmos/dbt/graph.py:438]: +1

**Review by pankajkoti — APPROVED** [review]
LGTM. minor questions inline

File: cosmos/dbt/graph.py (hunk 1 above)
Comment (tatiana) [reply cosmos/dbt/graph.py:438]: @pankajastro @pankajkoti MyPy didn't like it without it :( That's why I added it there. Do you have any suggestions

File: cosmos/dbt/graph.py (hunk 2 above)
Comment (tatiana) [reply cosmos/dbt/graph.py:445]: This line needs to be run regardless of caching - this how we're fixing the original issue

File: cosmos/dbt/graph.py (hunk 1 above)
Comment (pankajkoti) [reply cosmos/dbt/graph.py:438]: okay looks like MyPy is not able to figure out what's happening in the previous method. So I guess we can keep it for now until find a better resolution if any.

File: cosmos/dbt/graph.py (hunk 3 above)
Comment (tatiana) [reply cosmos/dbt/graph.py:482]: Resolving this as https://github.com/astronomer/astronomer-cosmos/pull/1070#discussion_r1657159419

**Review by pankajastro — APPROVED** [review]

**Pull request merged**